src/TerminalLinks.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2025-2025 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from "vscode";
+import * as fs from "fs";
+import { getChplHome } from "./configuration";
+
+export class ChplLink implements vscode.TerminalLink {
+  startIndex: number;
+  length: number;
+  data: { path: string; lineNum: number };
+
+  constructor(startIndex: number, length: number, path: string, lineNum: number) {
+    this.startIndex = startIndex;
+    this.length = length;
+    this.data = { path, lineNum };
+  }
+}
+
+// a terminal link provider to handle links like $CHPL_HOME/...
+export class ChplTerminalLinkProvider implements vscode.TerminalLinkProvider {
+  provideTerminalLinks(context: vscode.TerminalLinkContext, _token: vscode.CancellationToken): vscode.ProviderResult<ChplLink[]> {
+    const line = context.line;
+    const startIndex = line.indexOf("$CHPL_HOME");
+    if (startIndex === -1) {
+      return [];
+    }
+    const suffix = ".chpl";
+    const endIndex = line.indexOf(suffix + ":", startIndex);
+    if (endIndex === -1) {
+      return [];
+    }
+    var path = line.substring(startIndex, endIndex + suffix.length);
+
+    const lineNumStart = endIndex + suffix.length + 1;
+    const lineNumEnd = line.indexOf(":", lineNumStart);
+    const lineNum = lineNumEnd !== -1 ?
+      parseInt(line.substring(lineNumStart, lineNumEnd)) : -1;
```

Comment:
I'm not 100% sure that this is right, but it would be nice to add some NaN handling in case paseInt gets some weird value from this string